### PR TITLE
[4.x] Only allow POST requests for admin login

### PR DESCRIPTION
**Pull request by SniperSister**
### Summary of Changes
Enforce the usage of POST requests for administrator login requests – doesn't affect core CMS as it's doing POST logins by default anyway.

### Testing Instructions
Apply patch, login.

### Actual result BEFORE applying this Pull Request
Login works

### Expected result AFTER applying this Pull Request
Login works

### CODE REVIEW

**Comment by PhilETaylor** [comment]
This fixes the reported security issue. Im not sure why we are now resolving Joomla 4 security issues in public repeatedly now though. 

`'post'` is not needed. That is the default for the first param of `checkToken`

Did you take a wider look, or specific to the report? What about the other 13 uses of `checkToken('request')` - have you ensured those are also not introducing a security issue? (hint, there is another security issue

Should a GET have the ability to set the default "home" property for a list of items by GET request, when it should only be able to be set by POST (`@since 1.6`)

Surely the ones using `'request'` should be more specifically a "get" or a "post" (which is the default when no method set)? a method that can be called by GET and POST is probably badly coded I guess. 

`$this->checkToken('request');` is also used in the "Method to confirm the password request." - is that ok in context?

**Comment by zero-24** [comment]
> This fixes the reported security issue. Im not sure why we are now resolving Joomla 4 security issues in public repeatedly now though. 

Joomla 4 is not released stable yet and so a pre release. As long as the reported issue is an issue only within 4.x only or as in this case only will be fixed within 4.x it got an PR in the public tracker.

Reporting it to Security@ helps to check that distinction upfront for sure.

Edit: just to make that clear the above staremant is true until 4.x is in stable state. ;-)

File: administrator/components/com_login/src/Controller/DisplayController.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ public function display($cachable = false, $urlparams = false)
 	public function login()
 	{
 		// Check for request forgeries.
-		$this->checkToken('request');
+		$this->checkToken('post');
```
Comment (bembelimen) [inline administrator/components/com_login/src/Controller/DisplayController.php]:
```suggestion
		$this->checkToken();
```

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on ddd8f5cf1386afecf677b2669150b9ffdd5d4948<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31236">issues.joomla.org/tracker/joomla-cms/31236</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on ddd8f5cf1386afecf677b2669150b9ffdd5d4948<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31236">issues.joomla.org/tracker/joomla-cms/31236</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by PhilETaylor** [comment]
> Did you take a wider look, or specific to the report? What about the other 13 uses of checkToken('request') - have you ensured those are also not introducing a security issue? (hint, there is another security issue

Oh well, you can only lead a horse to water, you cannot make it drink... Im done.

**Pull request merged**